### PR TITLE
`_dynamic_info_firecrest_version` updated with new development

### PR DESCRIPTION
**Pull request by khsrali**
`/status/parameters` endpoint of firecrest api  is now updated with version value.
We fetch this way and drop the previous work around that was implemented.
Fixes:
https://github.com/aiidateam/aiida-firecrest/issues/43

### CODE REVIEW

File: tests/test_computer.py [hunk 1]
```diff
@@ -130,3 +131,28 @@ def test_dynamic_info(firecrest_config, monkeypatch, tmpdir: Path):
...
+        "api_version": f"{firecrest_config.api_version}",
+    }
+
+    # should catch FIRECREST_VERSION if value is not provided
+    result = _dynamic_info_firecrest_version(ctx, None, "0")
+    assert isinstance(result, str)
+
+    # should use the value if provided
+    result = _dynamic_info_firecrest_version(ctx, None, "10.10.10")
+    assert result == "10.10.10"
```
Comment (agoscinski) [inline tests/test_computer.py:158]: Maybe you can add tests for the raise of `BadParameter`

File: tests/test_computer.py [hunk 2]
```diff
@@ -107,9 +107,10 @@ def test_validate_temp_directory(
     ).exists()
 
 
-def test_dynamic_info(firecrest_config, monkeypatch, tmpdir: Path):
+def test_dynamic_info_direct_size(firecrest_config, monkeypatch, tmpdir: Path):
     from aiida_firecrest.transport import _dynamic_info_direct_size
 
+    assert monkeypatch is not None
```
Comment (agoscinski) [inline tests/test_computer.py:113]: Is this because of the linter? Seems a bit random

**Review by agoscinski — CHANGES_REQUESTED** [review]
Are [these comments](https://github.com/aiidateam/aiida-firecrest/blob/ff9b65ef54b9767085b9baa64707003bcd17bdb4/aiida_firecrest/transport.py#L132-L135) still up do date with this PR?

I am a bit confused how the function `_dynamic_info_firecrest_version` works. You specify a version by the value, but if it is 0 then it tries to to retrieve the  version dynamically. I guess that it is `0` and not `None` confuses me, but fine. Maybe can you add a bit doc to it because `value` is not such a meaningful name?

 Why is [the echo](https://github.com/aiidateam/aiida-firecrest/blob/ff9b65ef54b9767085b9baa64707003bcd17bdb4/aiida_firecrest/transport.py#L184-L186) only done in case it is dynamically retrieved? Maybe you can add a comment there.

**Comment by khsrali** [comment]
Thanks a lot, @agoscinski for the review!



> Are [these comments](https://github.com/aiidateam/aiida-firecrest/blob/ff9b65ef54b9767085b9baa64707003bcd17bdb4/aiida_firecrest/transport.py#L132-L135) still up do date with this PR?

yes, they are from the past, removed.

> 
> I am a bit confused how the function `_dynamic_info_firecrest_version` works. You specify a version by the value, but if it is 0 then it tries to to retrieve the version dynamically. I guess that it is `0` and not `None` confuses me, but fine. Maybe can you add a bit doc to it because `value` is not such a meaningful name?

Changed to `None`, doc added. Please check if it's clear. It's basically a click callback function.

> Why is [the echo](https://github.com/aiidateam/aiida-firecrest/blob/ff9b65ef54b9767085b9baa64707003bcd17bdb4/aiida_firecrest/transport.py#L184-L186) only done in case it is dynamically retrieved? Maybe you can add a comment there.

Just to prevent verbosity. Usually when inputs are valid during setting up/configuring a computer in aiida, it remains quite. 
I added echo for cases that it has to check something from the server. As a confirmation.

File: tests/test_computer.py [hunk 3]
```diff
@@ -150,9 +155,48 @@ def test_dynamic_info_firecrest_version(firecrest_config, monkeypatch, tmpdir: P
...
-    result = _dynamic_info_firecrest_version(ctx, None, "0")
+    result = _dynamic_info_firecrest_version(ctx, None, "None")
     assert isinstance(result, str)
 
     # should use the value if provided
     result = _dynamic_info_firecrest_version(ctx, None, "10.10.10")
     assert result == "10.10.10"
+
+    # raise BadParameter if the version is not supported
+    with pytest.raises(BadParameter) as exc_info:
```
Comment (agoscinski) [inline tests/test_computer.py:166]: pytest has an argument to match the error message
```suggestion
    with pytest.raises(BadParameter, match=r".*FirecREST api version 0.0.0 is not supported.*")):
```

File: tests/test_computer.py [hunk 4]
```diff
@@ -150,9 +155,48 @@ def test_dynamic_info_firecrest_version(firecrest_config, monkeypatch, tmpdir: P
...
+        result = _dynamic_info_firecrest_version(ctx, None, "0.0.0")
+    assert "FirecREST api version 0.0.0 is not supported" in str(exc_info.value)
```
Comment (agoscinski) [inline tests/test_computer.py:168]: see above
```suggestion
```

File: tests/test_computer.py [hunk 5]
```diff
@@ -130,3 +130,73 @@ def test_dynamic_info(firecrest_config, monkeypatch, tmpdir: Path):
...
+    result = _dynamic_info_firecrest_version(ctx, None, "10.10.10")
+    assert result == "10.10.10"
+
+    # raise BadParameter if the version is not supported
+    with pytest.raises(BadParameter) as exc_info:
+        result = _dynamic_info_firecrest_version(ctx, None, "0.0.0")
+    assert "FirecREST api version 0.0.0 is not supported" in str(exc_info.value)
+
+    # raise BadParameter if the input is nonsense, latest, stable, etc.
+    with pytest.raises(BadParameter) as exc_info:
```
Comment (agoscinski) [inline tests/test_computer.py:171]:
```suggestion
    with pytest.raises(BadParameter, match=r".* Invalid input.*"):
```

File: tests/test_computer.py [hunk 6]
```diff
@@ -130,3 +130,73 @@ def test_dynamic_info(firecrest_config, monkeypatch, tmpdir: Path):
...
+        result = _dynamic_info_firecrest_version(ctx, None, "latest")
+    assert "Invalid input" in str(exc_info.value)
```
Comment (agoscinski) [inline tests/test_computer.py:173]:
```suggestion
    with pytest.raises(BadParameter, match=r".*Invalid input.*"):
        _dynamic_info_firecrest_version(ctx, None, "latest")
```

File: tests/test_computer.py [hunk 7]
```diff
@@ -130,3 +130,73 @@ def test_dynamic_info(firecrest_config, monkeypatch, tmpdir: Path):
...
+    assert isinstance(result, str)
+
+    # should use the value if provided
+    result = _dynamic_info_firecrest_version(ctx, None, "10.10.10")
+    assert result == "10.10.10"
+
+    # raise BadParameter if the version is not supported
+    with pytest.raises(BadParameter) as exc_info:
+        result = _dynamic_info_firecrest_version(ctx, None, "0.0.0")
+    assert "FirecREST api version 0.0.0 is not supported" in str(exc_info.value)
```
Comment (agoscinski) [inline tests/test_computer.py:168]: pytest has an argument to match the error message
```suggestion
    with pytest.raises(BadParameter, match=r".*FirecREST api version 0.0.0 is not supported.*")):
       _dynamic_info_firecrest_version(ctx, None, "0.0.0")
```

**Comment by agoscinski** [comment]
Looks clear to me, I just simplified the pytest a little bit, check out my commit

File: tests/test_computer.py [hunk 8]
```diff
@@ -130,3 +130,71 @@ def test_dynamic_info(firecrest_config, monkeypatch, tmpdir: Path):
...
+    # should catch FIRECREST_VERSION if value is not provided
+    result = _dynamic_info_firecrest_version(ctx, None, "None")
+    assert isinstance(result, str)
+
+    # should use the value if provided
+    result = _dynamic_info_firecrest_version(ctx, None, "10.10.10")
+    assert result == "10.10.10"
+
+    # raise BadParameter if the version is not supported
+    with pytest.raises(BadParameter, match=r".*FirecREST api version 0.0.0 is not supported.*")):
```
Comment (agoscinski) [inline tests/test_computer.py:166]:
```suggestion
    with pytest.raises(BadParameter, match=r".*FirecREST api version 0.0.0 is not supported.*"):
```

**Comment by khsrali** [comment]
> Looks clear to me, I just simplified the pytest a little bit, check out my commit

Very good @agoscinski, thanks a lot!

**Pull request merged**